### PR TITLE
Region Selection bot: C# sample code

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
## Description

This bot has been created using [Bot Framework](https://dev.botframework.com), for the region selection using Card Bot and Cards.

## Testing
1. Ping the Bot
**Bot scenario:**

image
image
image

**Tab Scenario**
image
image
image

### CODE REVIEW

File: samples/app-region-selection/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+### Connect to the bot using Bot Framework Emulator
+
+- Launch Bot Framework Emulator
+- File -> Open Bot
+- Enter a Bot URL of `http://localhost:3978/api/messages`
+
+## Interacting with the bot
+
+Enter text in the emulator.  The text will be echoed back by the bot.
```
Comment (Wajeed-msft) [inline samples/app-region-selection/README.md:55]: Could you please add Microsoft Teams specific steps?

File: samples/app-region-selection/RegionSectionApp/Bots/RegionSelectionBot.cs [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+using Microsoft.BotBuilderSamples.Models;
+
+namespace Microsoft.BotBuilderSamples.Bots
+{
+    public class RegionSelectionTab : ActivityHandler
```
Comment (Wajeed-msft) [inline samples/app-region-selection/RegionSectionApp/Bots/RegionSelectionBot.cs:19]: Could you please check and remove if this is duplicate class?

File: samples/app-region-selection/RegionSectionApp/ConfigData/RegionDomains.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    {
+      "id": 3,
+      "region": "Asia & Pacific",
+      "country": "India",
+      "domain": ".in"
+    },
+    {
+      "id": 4,
+      "region": "Asia & Pacific",
```
Comment (Wajeed-msft) [inline samples/app-region-selection/RegionSectionApp/ConfigData/RegionDomains.json:23]: Could you please check if this repeat region?

File: samples/app-region-selection/RegionSectionApp/ConfigData/Regions.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+  "regionDomains": [
```
Comment (Wajeed-msft) [inline samples/app-region-selection/RegionSectionApp/ConfigData/Regions.json:2]: Could you please check which file is required and remove if unnecessary?

File: samples/app-region-selection/RegionSectionApp/DeploymentScripts/Linux/deploy.sh [hunk 5]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
```
Comment (Wajeed-msft) [inline samples/app-region-selection/RegionSectionApp/DeploymentScripts/Linux/deploy.sh:1]: Could you please remove DeploymentScripts folder?

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/app-region-selection/RegionSectionApp/ConfigData/RegionDomains.json (hunk 3 above)
Comment (Trinetra-MSFT) [reply samples/app-region-selection/RegionSectionApp/ConfigData/RegionDomains.json]: One is for India and other is for China region

File: samples/app-region-selection/README.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
+﻿# EchoBot
```
Comment (Wajeed-msft) [inline samples/app-region-selection/README.md:1]: Please correct this.

File: samples/app-region-selection/RegionSectionApp/Startup.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            // Create the User state.
+            services.AddSingleton<UserState>();
+
+            // Create the bot as a transient. In this case the ASP Controller is expecting an IBot.
+            services.AddTransient<IBot, RegionSelectionTab>();
+
+            services.AddMvcCore();
+                //.AddJsonFormatters(jsonOptions =>
+                //{
+                //    jsonOptions.DateTimeZoneHandling = DateTimeZoneHandling.Utc;
```
Comment (Wajeed-msft) [inline samples/app-region-selection/RegionSectionApp/Startup.cs:46]: Please remove commented code.

File: samples/app-region-selection/README.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  ```bash
+  # run the bot
+  dotnet run
+  ```
+
+  B) Or from Visual Studio
+
+  - Launch Visual Studio
+  - File -> Open -> Project/Solution
+  - Navigate to `samples/csharp_dotnetcore/DomainSelectionBot` folder
```
Comment (Wajeed-msft) [inline samples/app-region-selection/README.md:37]: Please use correct location.

File: samples/app-region-selection/README.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  - Select `EchoBot.csproj` file
+  - Press `F5` to run the project
+
+## Testing the bot using Bot Framework Emulator
```
Comment (Wajeed-msft) [inline samples/app-region-selection/README.md:41]: Please provide Microsoft Teams specific steps.

File: samples/app-region-selection/README.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+## Interacting with Region Selection Tab
+
+Set up the region selection app as a Tab in channel
+Select the region from the configuration page and click on save
+
+## Deploy the bot to Azure
+
+To learn more about deploying a bot to Azure, see [Deploy your bot to Azure](https://aka.ms/azuredeployment) for a complete list of deployment instructions.
+
+## Further reading
```
Comment (Wajeed-msft) [inline samples/app-region-selection/README.md:66]: Please give link to Microsoft Teams docs instead of Bot Framework docs.

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]
Please add link to this sample from root ReadMe.md file. Please add entry under scenario based samples.

**Comment by Trinetra-MSFT** [comment]
> Please add link to this sample from root ReadMe.md file. Please add entry under scenario based samples.

I will update the Root ReadMe file in other check-in i forgot to include it here so will do this with other

**Review by Wajeed-msft — APPROVED** [review]

**Review by asnow003 — APPROVED** [review]

**Pull request merged**